scripts/generate_dump

```diff
@@ -879,33 +879,47 @@ enable_logrotate() {
 collect_mellanox() {
     trap 'handle_error $? $LINENO' ERR
     local timeout_cmd="timeout --foreground ${TIMEOUT_MIN}m"
+    local platform=$(python3 -c "from sonic_py_common import device_info; print(device_info.get_platform())")
+    local hwsku=$(python3 -c "from sonic_py_common import device_info; print(device_info.get_hwsku())")
     local sai_dump_folder="/tmp/saisdkdump"
     local sai_dump_filename="${sai_dump_folder}/sai_sdk_dump_$(date +"%m_%d_%Y_%I_%M_%p")"
+    local sdk_dump_path=`cat /usr/share/sonic/device/${platform}/${hwsku}/sai.profile|grep "SAI_DUMP_STORE_PATH"|cut -d = -f2`
 
-    ${CMD_PREFIX}docker exec syncd mkdir -p $sai_dump_folder
-    ${CMD_PREFIX}docker exec syncd saisdkdump -f $sai_dump_filename
+    if [[ "$( docker container inspect -f '{{.State.Running}}' syncd )" == "true" ]]; then
+        if [[ x"$(sonic-db-cli APPL_DB EXISTS PORT_TABLE:PortInitDone)" == x"1" ]]; then
+            # Run saisdkdump only after the create_switch is known to be successful
+            ${CMD_PREFIX}docker exec syncd mkdir -p $sai_dump_folder
+            ${CMD_PREFIX}docker exec syncd saisdkdump -f $sai_dump_filename
 
-    if [ $? != 0 ]; then
-        echo "Failed to collect saisdkdump."
-    fi
+            if [ $? != 0 ]; then
+                echo "Failed to collect saisdkdump."
+            fi
 
-    copy_from_docker syncd $sai_dump_folder $sai_dump_folder
-    echo "$sai_dump_folder"
-    for file in `ls $sai_dump_folder`; do
-        save_file ${sai_dump_folder}/${file} sai_sdk_dump true
-    done
+            copy_from_docker syncd $sai_dump_folder $sai_dump_folder
+            echo "$sai_dump_folder"
+            for file in `ls $sai_dump_folder`; do
+                save_file ${sai_dump_folder}/${file} sai_sdk_dump true
+            done
 
-    ${CMD_PREFIX}rm -rf $sai_dump_folder
-    ${CMD_PREFIX}docker exec syncd rm -rf $sai_dump_folder
+            ${CMD_PREFIX}rm -rf $sai_dump_folder
+            ${CMD_PREFIX}docker exec syncd rm -rf $sai_dump_folder
+        fi
 
-    # Save SDK error dumps
-    local sdk_dump_path=`${CMD_PREFIX}docker exec syncd cat /tmp/sai.profile|grep "SAI_DUMP_STORE_PATH"|cut -d = -f2`
-    if [[ -d $sdk_dump_path ]]; then
+        # Save SDK error dumps
         copy_from_docker syncd $sdk_dump_path  /tmp/sdk-dumps
         for file in $(find /tmp/sdk-dumps -type f); do
             save_file ${file} sai_sdk_dump false
         done
         rm -rf /tmp/sdk-dumps
+    else
+        # Check if the SAI_DUMP_STORE_PATH is accessible on the host and collect files from there
+        if [[ -d $sdk_dump_path ]]; then
+            $CP -rvf $sdk_dump_path /tmp/sdk-dumps
+            for file in $(find /tmp/sdk-dumps -type f); do
+                save_file ${file} sai_sdk_dump false
```

Comment:
We may need to remove this logic as now techsupport will contain the same saisdkdumps in two paths one under /var/log and another under sai_sdk_dump

Reply:
We collect new dumps under `/var/log/orch_abrt_status`, fwdump me collects under SAI_DUMP_STORE_PATH i.e. $sdk_dump_path. So, they won't be same.

Reply:
No. What I meant was there will be two paths due to save_file being called.
By default everything under /var/log will be collected
The save_file will create a new directory and copy the dump file which means techsupport will have same dump file under two paths.
1) Remove this save_file logic or
2) After doing save file, delete the file under /var/log

Reply:
Only individual files under /var/log are collected by default. https://github.com/vivekrnv/sonic-utilities/blob/4bce0208da1b6f9c7cfbd23c5f931f7b8bc1b41b/scripts/generate_dump#L838 So, there won't be two paths

Reply:
Fixed. Please review